topic-controller/README.md

```diff
@@ -62,6 +60,92 @@ then Kafka itself cannot run, so the controller will be no less available
 than it would even if it was stateless. 
 
 
+## Usage Recommendations
+
+1. Decide to either always operate on ConfigMaps or always operate directly on topics.
+2. When creating a ConfigMap:
+    * Remember that you can't easily change the name later.
+    * Choose a name for the ConfigMap that reflects the name of the topic it describes.
+    * Ideally the ConfigMap's `metadata.name` should be the same as its `data.name`.
+3. When creating a topic:
+    * Remember that you can't change the name later.
+    * It's best to use a name that is a [valid Kubernetes resource name][identifiers], 
+      otherwise the controller will have to sanitize the name when creating 
+      the corresponding ConfigMap.
+    
+## Format of the ConfigMap
+
+By default, the controller only considers ConfigMaps having the label `strimzi.io/kind=topic`, 
+but this is configurable via the `TC_CM_LABELS` environment variable.
+
+The `data` of such ConfigMaps supports the following keys:
+
+* `name` The name of the topic. If this is absent the name of the ConfigMap itself is used.
+* `partitions` The number of partitions of the Kafka topic. This can be increased, but not decreased.
+* `replicas` The number of replicas of the Kafka topic. 
+* `config` A string in JSON format representing the topic configuration. 
+
+## Example
+
+Suppose you want to create a topic called "orders" with 10 partitions and 2 replicas. 
+
+You would first prepare a ConfigMap:
+
+```
+# contents of orders-topic.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: orders
+  labels:
+    kind: strimzi.io/kind=topic
+data:
+  name: orders
+  partitions: "10"
+  replicas: "2"
+  config: '{}' 
```

Comment:
If the user can omit `config` then at least you should explain that this field is optional and the default topic configuration will be used. Of course it could be true even for the other fields (`partitions`, `replicas`, ...)